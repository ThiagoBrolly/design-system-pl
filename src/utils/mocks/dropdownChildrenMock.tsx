export const dropdownChildrenMock = (
  <>
    <p>Item 1</p>
    <span>Item 2 ...........................</span>
    <p>Item 3</p>
    <p>Item 4</p>
    <p>Item 5</p>
    <p>Item 6</p>
    <p>Item 7</p>
    <p>Item 8</p>
    <p>Item 9</p>
    <p>Item 10</p>
    <p>Item 11</p>
    <p>Item 12</p>
    <p>Item 13</p>
    <p>Item 14</p>
    <p>Item 15</p>
    <p>Item 16</p>
    <p>Item 17</p>
    <p>Item 18</p>
  </>
);
