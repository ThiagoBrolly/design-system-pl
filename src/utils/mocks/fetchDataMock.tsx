export const searchUserDataMock = [
  'Manoel Pereira',
  'Diego Arend',
  'Hugo Gitz',
  'Derickson Loss',
  'Thiago Sousa',
  'Tatiani da Silva',
  'Mateus Junges',
  'Matheus Morais',
  'Mateus Junges2',
  'Matheus Morais2',
  'Mateus Junges3',
  'Matheus Morais3',
  'Mateus Junges4',
  'Matheus Morais4',
  'Mateus Junges5',
  'Matheus Morais5',
  'Mateus Junges6',
  'Matheus Morais6',
  'Mateus Junges xxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxx xxxxxxxxxxxxx',
  'Matheus Morais7',
  'Mateus Junges7',
  'Matheus Morais8',
  'Mateus Junges9',
  'Matheus Morais9',
  'Mateus Junges10',
  'Matheus Morais10',
];

export const fetchDataMock = (searchPhrase: string) => {
  const searchResult = searchUserDataMock.filter((option) =>
    option.toLowerCase().includes(searchPhrase.toLowerCase()),
  );
  if (searchResult.length > 0) {
    return true;
  }
  return false;
};

export const childrenMock = (
  <>
    <p>
      Sucesso. Aqui deve-se passar o children din√¢mico e corretamente filtrado
    </p>
  </>
);

// export const searchUserDataMock = [
//   { id: '01', name: 'Manoel Pereira', email: 'manoel@paylivre.com' },
//   'Diego Arend',
//   'Hugo Gitz',
//   'Derickson Loss',
//   'Thiago Sousa',
//   'Tatiani da Silva',
//   'Mateus Junges',
//   'Matheus Morais',
//   'Mateus Junges2',
//   'Matheus Morais2',
//   'Mateus Junges3',
//   'Matheus Morais3',
//   'Mateus Junges4',
//   'Matheus Morais4',
//   'Mateus Junges5',
//   'Matheus Morais5',
//   'Mateus Junges6',
//   'Matheus Morais6',
//   'Mateus Junges xxxxxxxxxxxxxxxxxxxx xxxxxxxxxxxxxxxxx xxxxxxxxxxxxx',
//   'Matheus Morais7',
//   'Mateus Junges7',
//   'Matheus Morais8',
//   'Mateus Junges9',
//   'Matheus Morais9',
//   'Mateus Junges10',
//   'Matheus Morais10',
// ];
