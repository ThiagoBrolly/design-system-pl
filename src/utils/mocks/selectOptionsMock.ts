// export const selectOptionsMock = [
//   { value: 'One item', displayValue: 'One item' },
//   { value: 'Two itens', displayValue: 'Two itens' },
//   { value: 'Three itens', displayValue: 'Three itens' },
//   { value: 'Four itens', displayValue: 'Four itens' },
//   { value: 'Five itens', displayValue: 'Five itens' },
// ];

export const selectOptionsMock = [
  { value: 1, displayValue: 'One item' },
  { value: 2, displayValue: 'Two itens' },
  { value: 3, displayValue: 'Three itens' },
  { value: 4, displayValue: 'Four itens' },
  { value: 5, displayValue: 'Five itens' },
];

// export const selectOptionsMock = [
//   'One item',
//   'Two itens',
//   'Three itens',
//   'Four itens',
//   'Five itens',
// ];
