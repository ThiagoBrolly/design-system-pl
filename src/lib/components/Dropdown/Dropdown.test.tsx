import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Dropdown } from '.';
import { dropdownChildrenMock } from '../../../utils/mocks/dropdownChildrenMock';

describe('<Dropdown />', () => {
  test('should correctly render the dropdown component', () => {
    const { container } = render(<Dropdown title="title test" />);

    expect(screen.getByRole('menu')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByLabelText('dropdown-title')).toHaveTextContent(
      'title test',
    );
  });
  test('should correctly open/close the dropdown menu (after clicking)', () => {
    render(<Dropdown children={dropdownChildrenMock} />);
    const dropdownHeader = screen.getByLabelText('dropdown-header');
    const dropdownOverlay = screen.getByLabelText('dropdown-overlay');
    // open - shows the dropdown list
    userEvent.click(dropdownHeader);
    let dropdownList = screen.getByLabelText('dropdown-list');
    expect(dropdownList).toBeInTheDocument();
    // persist list when clicking on it
    userEvent.click(dropdownList);
    expect(dropdownList).toBeInTheDocument();
    // close - hides the dropdown list
    userEvent.click(dropdownHeader);
    expect(dropdownList).not.toBeInTheDocument();
    // close - hides the dropdown list (when clicking outside - overlay)
    userEvent.click(dropdownHeader);
    dropdownList = screen.getByLabelText('dropdown-list');
    expect(dropdownList).toBeInTheDocument();
    userEvent.click(dropdownOverlay);
    expect(dropdownList).not.toBeInTheDocument();
  });
});
