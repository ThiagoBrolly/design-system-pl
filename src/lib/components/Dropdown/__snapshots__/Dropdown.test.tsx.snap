// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should correctly render the dropdown component 1`] = `
.c2 {
  position: relative;
  height: 56px;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 32rem;
  background-color: #ffffff;
  border-radius: 1rem;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
  cursor: pointer;
  z-index: 50;
}

.c4 {
  height: 24px;
  font-size: 2rem;
  margin-right: 16px;
  cursor: pointer;
  color: #a8a8a8;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  line-clamp: 1;
  -webkit-box-orient: vertical;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 30;
}

.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c0 .c1 {
  border: 0.1rem solid #b1b8be;
}

.c0 .c1:hover {
  border: 0.1rem solid #0f194b;
}

<div
  class="c0"
>
  <div
    aria-label="dropdown-container"
    class="c1 c2"
    role="menu"
  >
    <div
      aria-label="dropdown-header"
      class="c3"
    >
      <title
        aria-label="dropdown-title"
        class="c4"
      >
        title test
      </title>
      <svg
        fill="#215aff"
        viewBox="0 0 21.3 21.3"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.811 21.067 15.847 17.1a10.129 10.129 0 0 1-3.378 1.35 8.8 8.8 0 0 1-5.387-.528c-2.326-.977-4.417-2.8-6.454-5.938A4.717 4.717 0 0 1 0 10.733c0-.676 1.179-2.233 1.7-2.928a23.087 23.087 0 0 1 2.365-2.483l-3.8-3.8A.889.889 0 0 1 1.517.26l4.425 4.425a.676.676 0 0 1 .051.038.8.8 0 0 1 .156.169L9.4 8.139l.024.024 7.132 7.137.029.029 4.482 4.482a.888.888 0 1 1-1.256 1.256ZM3.985 7.873a25.257 25.257 0 0 0-2.1 2.767 12.963 12.963 0 0 0 5.885 5.648 7.3 7.3 0 0 0 4.244.482 6.718 6.718 0 0 0 2.533-.967l-2.061-2.061a3.555 3.555 0 0 1-4.9-4.9L5.324 6.581c-.363.319-.829.751-1.339 1.292Zm4.89 3.3a1.777 1.777 0 0 0 2.284 1.247l-2.248-2.252a1.776 1.776 0 0 0-.036 1.002Zm9.1 3.03a.889.889 0 0 1-.13-1.25 36.17 36.17 0 0 0 1.64-2.211 13.018 13.018 0 0 0-5.865-5.686 7.341 7.341 0 0 0-4.6-.43.888.888 0 1 1-.431-1.726 9.126 9.126 0 0 1 5.72.516c2.326.977 4.4 2.842 6.433 5.974a4.848 4.848 0 0 1 .6 1.259 8.2 8.2 0 0 1-1.332 2.411c-.346.469-.755.974-.783 1.009a.888.888 0 0 1-1.25.13Z"
          data-name="Caminho 3829"
        />
      </svg>
    </div>
  </div>
  <div
    aria-hidden="true"
    aria-label="dropdown-overlay"
    class="c5"
  />
</div>
`;
